test/unit/host_test.rb

```diff
@@ -2641,6 +2641,12 @@ def to_managed!
 
       assert_equal actual_attr['zzz_id'], 1111
     end
+
+    test 'should add inherited attributes when hostgroup in attributes' do
+      hg = FactoryGirl.create(:hostgroup, :with_puppetclass)
```

Comment:
If you can only set the environment ID on the host group (perhaps refactor :with_puppetclass into a new :with_environment trait) then it'll reduce test DB load.